src/Service/Startup.cs

```diff
@@ -798,8 +798,16 @@ private void ConfigureAuthentication(IServiceCollection services, RuntimeConfigP
                         }
                     }
 
-                    services.AddAuthentication(EasyAuthAuthenticationDefaults.AUTHENTICATIONSCHEME)
+                    string scheme = easyAuthType switch
+                    {
+                        EasyAuthType.AppService => EasyAuthAuthenticationDefaults.APPSERVICEAUTHSCHEME,
+                        EasyAuthType.StaticWebApps => EasyAuthAuthenticationDefaults.SWAAUTHSCHEME,
+                        _ => EasyAuthAuthenticationDefaults.AUTHENTICATIONSCHEME
+                    };
+
+                    services.AddAuthentication(scheme)
                         .AddEasyAuthAuthentication(easyAuthAuthenticationProvider: easyAuthType);
+                    _logger.LogInformation($"Registered EasyAuth scheme: {scheme}");
                 }
```

Comment:
As per the Description, `AddEnvDetectedEasyAuth` should be modified to add AppService irrespective of detection of the AppService environment. But that function has not been modified. We should fix that function to depend on the boolean `appServiceEnvironmentDetected`, instead always add the AppService provider.